### ISSUE TITLE
Enhance auto-defer in `ui.View` to handle long-running callbacks before response

### ISSUE THREAD

**Issue by drlove2002** [nextcord/nextcord#1227]
### Summary

Improve the existing auto-defer mechanism in ui.View to account for cases where a callback execution takes more than Discord's 3-second interaction response timeout. This enhancement would ensure deferrals occur proactively during the callback execution period, reducing errors and improving usability.

### What is the feature request for?

The core library

### The Problem

Currently, the auto-defer functionality in `ui.View` only defers the response after a callback has completed. If the callback takes longer than 3 seconds to execute (e.g., due to API calls, database queries, or complex processing), Discord may display the "This interaction failed" error before the auto-defer logic triggers.

This logic defers after the callback finishes:
```py
async def _scheduled_task(self, item: Item, interaction: Interaction):
    ...
    await item.callback(interaction)
    if (
        not interaction.response._responded
        and not interaction.is_expired()
        and self.auto_defer
    ):
        await interaction.response.defer()
```
However, if the callback exceeds 3 seconds, the defer will be too late to prevent a failure.

### The Ideal Solution

Enhance the current auto-defer mechanism to defer proactively **during** the callback execution. This can be achieved by:

1. Scheduling a deferral task that runs concurrently with the callback:
   - If the callback exceeds the timeout threshold (e.g., 2.7 seconds), the response is deferred automatically.
   - If the callback completes early and sends a response, the deferral task is canceled.

2. Adding an optional `defer_time` parameter to `ui.View` for developers to configure the proactive defer timeout.

Example enhancement:
```python
class View(ui.View):
    def __init__(self, auto_defer: bool = True, timeout: int = 180, defer_time: float = 2.7):
        super().__init__(timeout=timeout)
        self.auto_defer = auto_defer
        self.defer_time = defer_time

    async def _defer_task(self, interaction: Interaction):
        await asyncio.sleep(self.defer_time)
        if not interaction.response._responded and not interaction.is_expired():
            await interaction.response.defer()

    async def _scheduled_task(self, item: ui.Item, interaction: Interaction):
        defer_task = (
            asyncio.create_task(self._defer_task(interaction))
            if self.auto_defer
            else None
        )
        try:
            if self.timeout:
                self.__timeout_expiry = time.monotonic() + self.timeout

            allow = await self.interaction_check(interaction)
            if not allow:
                return None

            await item.callback(interaction)
        except Exception as e:
            return await self.on_error(e, item, interaction)
        finally:
            if defer_task and not defer_task.done():
                defer_task.cancel()
```

### The Current Solution

Currently, the auto-defer logic in `ui.View` defers responses **only after** the callback execution. If the callback finishes within Discord's 3-second timeout, the defer works fine. However, if the callback execution exceeds this time, the response will fail before the defer can occur, resulting in a "This interaction failed" error.

For example, this is the current behavior:
```python
async def _scheduled_task(self, item: Item, interaction: Interaction):
    try:
        if self.timeout:
            self.__timeout_expiry = time.monotonic() + self.timeout

        allow = await self.interaction_check(interaction)
        if not allow:
            return None

        await item.callback(interaction)  # Executes the callback
        if (
            not interaction.response._responded
            and not interaction.is_expired()
            and self.auto_defer
        ):
            await interaction.response.defer()  # Defers only after the callback
    except Exception as e:
        return await self.on_error(e, item, interaction)
```

To mitigate this, developers currently need to handle deferrals **manually** within their callbacks, especially for tasks that might exceed 3 seconds. This involves explicitly checking and deferring the interaction early on, like this:

```python
async def button_callback(interaction: Interaction):
    if not interaction.response.is_done():  # Check if response is pending
        await interaction.response.defer()  # Manually defer
    await asyncio.sleep(3)  # Simulate a long task
    await interaction.response.send_message("Task completed!")
```

This approach works but requires repetitive code and manual intervention for every potentially long-running callback, which could be handled automatically by the framework.

### Additional Context

- This improvement would enhance the developer experience by ensuring consistent and automatic handling of interaction timeouts.
- Fits naturally into Nextcord’s existing abstraction for `ui.View`.
- The enhancement would maintain backward compatibility while improving reliability for complex bots.
